packages/components/src/components/stepper-horizontal/stepper-horizontal/stepper-horizontal.tsx

```diff
@@ -133,33 +118,33 @@ export class StepperHorizontal {
     );
   }
 
-  private addEventListeners = (): void => {
-    // TODO: why not apply via jsx?
-    this.scrollerElement.addEventListener('click', (e) => {
-      const target = getClickedItem<HTMLPStepperHorizontalItemElement>(
-        this.host,
-        'p-stepper-horizontal-item',
-        e.composedPath()
-      );
+  private onClickScroller = (e: MouseEvent): void => {
+    const target = getClickedItem<HTMLPStepperHorizontalItemElement>(
+      this.host,
+      'p-stepper-horizontal-item',
+      e.composedPath()
+    );
 
-      if (target) {
-        const clickedStepIndex = this.stepperHorizontalItems.indexOf(target);
+    if (target) {
+      const clickedStepIndex = this.stepperHorizontalItems.indexOf(target);
 
-        this.update.emit({ activeStepIndex: clickedStepIndex });
-        this.stepChange.emit({ activeStepIndex: clickedStepIndex });
-      }
-    });
+      this.update.emit({ activeStepIndex: clickedStepIndex });
+      this.stepChange.emit({ activeStepIndex: clickedStepIndex });
+    }
+  };
+
+  private createStepperHorizontalItemElements = (): HTMLPStepperHorizontalItemElement[] => {
+    return Array.from(this.host.children) as HTMLPStepperHorizontalItemElement[];
   };
 
   private defineStepperHorizontalItemElements = (): void => {
-    // TODO: validation? this could be any kind of dom node
-    this.stepperHorizontalItems = Array.from(this.host.children) as HTMLPStepperHorizontalItemElement[];
+    this.stepperHorizontalItems = this.createStepperHorizontalItemElements();
   };
 
   private validateComponent = (): void => {
     throwIfChildrenAreNotOfKind(this.host, 'p-stepper-horizontal-item');
     throwIfChildCountIsExceeded(this.host, 9);
-    throwIfMultipleCurrentStates(this.host, this.stepperHorizontalItems);
+    throwIfMultipleCurrentStates(this.host, this.createStepperHorizontalItemElements());
```

Comment:
This got a little bit confusing and tough to minify since those class functions can't be minified. 
Why don't we drop those 1 line utils within the class and instead do this
```
this.stepperHorizontalItems = this.createStepperHorizontalItemElements();
```
as part of `validateComponent()`. 

That would be cleaner than doing those 2 steps after each other multiple times
```
this.validateComponent();
this.defineStepperHorizontalItemElements();
```